### PR TITLE
Correct the logic for when bill split particpants can be selected

### PR DESCRIPTION
**Pull request by tgolen**
cc @luacmartins since it appears you flipped the logic in this PR: https://github.com/Expensify/App/pull/11597#issuecomment-1297711122. I was talking with @trjExpensify about it and that was a mistake.

### Fixed Issues
$ https://github.com/Expensify/App/issues/13220

##### Test the floating-action-button
1. Click the floating action button
2. Select Split Bill
3. Enter an amount and click Next
4. Select multiple people and click Next
5. Verify that at the final confirmation screen, you CANNOT select or deselect anyone

##### Test the in-chat menu
1. Go to a group chat
2. Click the "+" button attached to the comment composer
3. Select Split bill
4. Enter an amount and click Next
5. Verify that at the final confirmation screen, you CAN select and deselect people

- [ ] Verify that no errors appear in the JS console

### Offline tests
Same as the above tests, it should behave the same

### QA Steps
Same tests as above

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/1228807/205361338-efc1c17a-6d16-4289-9dfa-4b6dbbeeb537.mp4

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/1228807/205361355-f07c4522-2c5f-4157-83eb-bfd2f3d6c524.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/1228807/205361377-205a4f94-3876-4e30-ac2e-9bca9f92660a.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/1228807/205361396-265abbc7-ca0b-4ac7-ba76-8ca49542d2b5.mp4

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/1228807/205361411-8e8aca8e-bb3f-4c68-91d6-c61cc789199c.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/1228807/205361423-ef5958f2-73e5-4d6e-86e8-56e432ad0e20.mp4

</details>

### CODE REVIEW

File: src/components/IOUConfirmationList.js [hunk 1]
```diff
@@ -57,7 +57,8 @@ const propTypes = {
         phoneNumber: PropTypes.string,
     })).isRequired,
 
-    /** Is this IOU associated with existing report */
+    /** Is this IOU associated with existing report. This is true when the action is initiated from inside a group chat and it's false when the action is initiated from the
+     * floating-action-button */
     isIOUAttachedToExistingChatReport: PropTypes.bool.isRequired,
```
Comment (tgolen) [inline src/components/IOUConfirmationList.js:62]: There is a part of me that wants to rename this prop to `canModifyParticipants` and make it more clear what's happening. Let me know if you agree.

**Comment by luacmartins** [comment]
@tgolen I asked @JmillsExpensify about this change [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1666899921778839?thread_ts=1666670872.559659&cid=C01GTK53T8Q) and it seems like the DD mentions not being able to adjust members on this page. I'm happy to change if that's the intended behavior though

**Comment by JmillsExpensify** [comment]
Should we take this back to Slack?

**Comment by trjExpensify** [comment]
It was changed later than the original release of N5 as a follow-up, so I imagine that's why it isn't in the OG doc when you looked there. It was changed to allow deselecting in a groupDM because sometimes not everyone in the group is part of every single shared expense, like if you created a group for a trip you were all going on a la splitwise. Else, you would have to create all these sub groups for when someone skips a dinner etc.

File: src/components/IOUConfirmationList.js (hunk 1 above)
Comment (sketchydroide) [reply src/components/IOUConfirmationList.js:62]: yeah the name is hard to read and I don't understand what it supposed to be right now, your suggestion makes sense

Comment (sketchydroide) [reply src/components/IOUConfirmationList.js:62]: if it's the only thing that it's doing right now

**Review by sketchydroide — DISMISSED** [review]
The code is good as it is, but feel free to do the changes on the props name, will review again

**Review by tgolen — COMMENTED** [review]
OK, sounds like we are settled with this being the correct logic. 

@sketchydroide I have renamed the prop and moved the location of the detailed comment as well. Thanks!

**Review by sketchydroide — APPROVED** [review]
Yeah this makes more sense now

**Comment by sketchydroide** [comment]
@sobitneupane happy to merge once you do your review

**Review by luacmartins — APPROVED** [review]
LGTM!

**Comment by sobitneupane** [comment]
Noticed following things while testing the PR. Just noting down to make sure they are expected.
- When we split bill from a group with three members (excluding user) and remove a member from IOU confirmation page, then new group with two members gets created with split message.
- When we split bill from a group and remove all other members expect one from IOU confirmation page, then no split message appears any where. Only money requested message in one-to-one chat.
- When we split bill from a room, new group (with members from the room) gets created with split message.

**Review by sobitneupane — COMMENTED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/25876548/205901851-c484a257-0075-4a94-bb78-94ffb9507276.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/25876548/205901925-446f1e46-7b5d-47c3-a310-8b8656a629b3.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/25876548/205901995-7d6f383d-2bba-42fa-b81e-b37c0dae44d9.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/25876548/205901901-c8cb20b5-b65b-4d10-a2fa-7b5c17e8610b.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/25876548/205902213-1fbe8401-c2b8-466a-9ed5-d577ad41832d.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/25876548/205902198-b724e7df-d942-4884-8ca1-2b0716d7a0f4.mov


</details>

**Review by sobitneupane — APPROVED** [review]
Changes look good and test well. Noted few things [here](https://github.com/Expensify/App/pull/13279#issuecomment-1339167263) just to make sure they are expected.

🎀👀🎀 C+ reviewed

cc: @sketchydroide @luacmartins

**Comment by trjExpensify** [comment]
>When we split bill from a group with three members (excluding user) and remove a member from IOU confirmation page, then new group with two members gets created with split message.

This isn't expected. The split message should go in the existing groupDM between all members. For example:

1. As memberA, create a groupDM with memberB, memberC & memberD
2. Go to the `+` menu inside the groupDM > click `Split bill`
3. Enter $100 > click `Next`
4. Deselect memberD > click `Split $100`
5. The split action from memberA is added to the same groupDM: `Split $100 with memberB and memberC`
6. In the memberA:memberB DM the request action from memberA is added: `Requested $33.33 from memberB`
7. In the memberA:memberC DM the request action from memberA is added: `Requested $33.33 from memberC`
8. In the memberA:memberD DM there is nothing added, as memberD was excluded from the split. 

>When we split bill from a room, new group (with members from the room) gets created with split message.

This isn't expected. Just to ensure we're on the same page though about why, is this what's happening?

1. Create a groupDM with memberA, memberB, memberC, such that one exists between these three members
2. Go to the `+` menu inside the groupDM and split a bill with all members
3. A new groupDM gets created between userA, userB, userC such that there's now two groupDMs with exactly the same three members

DMs should always be unique. There can't be two DMs with the same members, whether that be a 1:1 DM or a groupDM. 

>When we split bill from a group and remove all other members expect one from IOU confirmation page, then no split message appears any where. Only money requested message in one-to-one chat.

This isn't expected either. If you're inside a groupDM and splitting a bill, the split message should be posted in that groupDM you took the action from, as that's your intention for visibility for the other group members, else you wouldn't be going through this flow? Writing it out for completeness:

1. As memberA, create a groupDM with memberB and memberC
2. Go to the `+` menu inside the groupDM > click `Split bill`
3. Enter $100 > click `Next`
4. Deselect memberC > click `Split $100`
5. The split action from memberA is added to the same groupDM: `Split $100 with memberB`
6. In the memberA:memberB DM the request action from memberA is added: `Requested $50.00 from memberB`
7. In the memberA:memberC DM there is nothing added, as memberC was excluded from the split.

**Comment by sobitneupane** [comment]
> > When we split bill from a room, new group (with members from the room) gets created with split message.
> 
> This isn't expected. Just to ensure we're on the same page though about why, is this what's happening?
> 
> 1. Create a groupDM with memberA, memberB, memberC, such that one exists between these three members
> 2. Go to the `+` menu inside the groupDM and split a bill with all members
> 3. A new groupDM gets created between userA, userB, userC such that there's now two groupDMs with exactly the same three members

@trjExpensify When we split bill from a room (not group), group gets created with split message.

1. Create a room with memberA, memberB, memberC
2. Go to the `+` menu inside the room and split a bill with all members
3. A new groupDM gets created between userA, userB, userC

**Comment by luacmartins** [comment]
>When we split bill from a group with three members (excluding user) and remove a member from IOU confirmation page, then new group with two members gets created with split message.

Interesting. I'm pretty sure this was the behavior before the refactors and I intentionally refactored the new command to do this. It seems like we need to change this based on @trjExpensify's comments?

>When we split bill from a room, new group (with members from the room) gets created with split message.

@trjExpensify isn't this expected since these are IOUs (P2P) instead of an expense report? Do we wanna keep personal requests in policy rooms?

>When we split bill from a group and remove all other members expect one from IOU confirmation page, then no split message appears any where. Only money requested message in one-to-one chat.

This seems like a bug. The current behavior adds both messages to a split request with just one other user.

**Comment by tgolen** [comment]
I love all the comments about that, but would it be OK to move this outside of the PR? This PR is specifically about flipping the logic for when participants can be modified so all of this is out-of-scope.

**Pull request merged**